### ISSUE TITLE
Stumbled upon some CF documentation that might be of some possible interest

### ISSUE THREAD

**Issue by mhlslash** [hercules-390/hyperion#142]
I came across what looks to be some interesting documentation regarding the CF and CFCC -

https://www-05.ibm.com/e-business/linkweb/publications/servlet/pbi.wss?CTY=US&FNC=SRX&PBL=SA22-7618-15

I'm not sure if that's anything useful or not, I do know that there's been talk on the hercules-390 list about interest in implementing a CF emulation within Hercules, and that the general response has been that there hasn't been much documentation found on it. Again, this info might be completely useless, but just something that I came across that I thought I'd at least bring up in case it is of any use/help in implementing a possible emulation.

**Comment by Fish-Git** [comment]
> I came across what looks to be some interesting documentation regarding the CF and CFCC -
> 
> https://www-05.ibm.com/e-business/linkweb/publications/servlet/pbi.wss?CTY=US&FNC=SRX&PBL=SA22-7618-15

SA22-7618-17 zOS MVS Programming - Sysplex Services Reference:

_**Note: Before using this information and the product it supports, read the information in “Notices” on page v.**_
[...]
**_NOTICES:_**
[...] _**IBM may have patents or pending patent applications covering subject matter described in this document. The furnishing of this document does not give you any license to these patents.**_ [...]

> [...] and that the general response has been that there hasn't been much documentation found on it.

I don't believe so.  AFAIK the problem has been (and still is) the aforementioned patent issue.

> [...] in case it is of any use/help in implementing a possible emulation.

For what it's worth, the very first commit of Hercules (b95ef23f81f6220bdd9bf65e64d07911b57d89d3) _did_ contain CF support code (search for `FEATURE_HARDWARE_LOADER`), but it was quickly ripped out in the very next commit due to, as I recall, the patent/licensing issues involved.

It's been a while since I looked at it but as I recall Jan's code did not actually emulate the CF code but rather simply supported the loading and executing of IBM's Coupling Facility licensed internal code itself that you had to acquire yourself elsewhere. (How it did that I don't know.)

Bottom line, I _seriously_ doubt you'll see any type of CF code in Hercules.

At least not in our lifetime.
